Sources/WalletConnectNotify/Client/Wallet/ProtocolEngine/wc_notifyWatchSubscriptions/NotifyWatchSubscriptionsResponseSubscriber.swift

```diff
@@ -38,17 +38,19 @@ class NotifyWatchSubscriptionsResponseSubscriber {
                 let (watchSubscriptionPayloadRequest, _) = try NotifyWatchSubscriptionsPayload.decodeAndVerify(from: payload.request)
 
                 let account = watchSubscriptionPayloadRequest.subscriptionAccount
-                // todo varify signature with notify server diddoc authentication key
+                // TODO: varify signature with notify server diddoc authentication key
 
-                let oldSubscriptions = notifyStorage.getSubscriptions()
+                let oldSubscriptions = notifyStorage.getSubscriptions(account: account)
                 let newSubscriptions = try await notifySubscriptionsBuilder.buildSubscriptions(responsePayload.subscriptions)
 
                 try Task.checkCancellation()
 
-                logger.debug("number of subscriptions: \(newSubscriptions.count)")
+                let subscriptions = oldSubscriptions.difference(from: newSubscriptions)
```

Comment:
I just wonder how much more difficult it would be to change the type of subscriptions to a Set<Subscriptions>

Reply:
inside difference extension it's casting arrays to Sets